test/integration/datasets/DatasetsRepository.test.ts

```diff
@@ -95,6 +96,45 @@ describe('DatasetsRepository', () => {
         `There was an error when reading the resource. Reason was: [404] Dataset with ID ${nonExistentTestDatasetId} not found.`,
       );
     });
+    test('should return citation when dataset is deaccessioned', async () => {
+      let createdDatasetId = undefined;
+      await createDatasetViaApi()
+          .then((response) => (createdDatasetId = response.data.data.id))
+          .catch(() => {
+            assert.fail('Error while creating test Dataset');
+          });
+      // We publish the new test dataset we can deaccession it
+      await publishDatasetViaApi(createdDatasetId)
+          .then()
+          .catch(() => {
+            assert.fail('Error while publishing test Dataset');
+          });
+
+      let locks = await sut.getDatasetLocks(createdDatasetId);
+      const maxTries = 10;
+      let tries = 0;
+      while (locks.length > 0 && tries < maxTries) {
+            await new Promise(resolve => setTimeout(resolve, 1000));
+            locks = await sut.getDatasetLocks(createdDatasetId);
+            tries++
+      }
+      if (tries >= maxTries && locks.length > 0) {
+        assert.fail('Error while waiting for locks to be released');
+      }
```

Comment:
To improve the readability I think we should move the locks check loop to the datasetHelper and just import the checkLocks function here

Reply:
I created a method in datasetHelper to wait for locks, but I commented it out, because it was causing this jest error:
Node.js v20.6.1
node:internal/child_process/serialization:159
    const string = JSONStringify(message) + '\n';
                   ^

TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'Object'
    |     property 'res' -> object with constructor 'Object'
    --- property 'req' closes the circle
    at stringify (<anonymous>)
    at writeChannelMessage (node:internal/child_process/serialization:159:20)
    at process.target._send (node:internal/child_process:852:17)
    at process.target.send (node:internal/child_process:752:19)
    at reportSuccess (/Users/ellenkraffmiller/Documents/GitHub/dataverse-client-javascript/node_modules/jest-worker/build/workers/processChild.js:82:11)
    
The call seems to be generated from the axios.get() method.   I'm not sure why this is happening here, and not in the other calls to getDatasetLocks.